common/src/main/scala/explore/utils/TableHooks.scala

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.utils
+
+import cats.effect.IO
+import cats.syntax.all.*
+import crystal.react.ReuseView
+import crystal.react.View
+import crystal.react.hooks.*
+import crystal.react.implicits.*
+import crystal.react.reuse.*
+import explore.common.UserPreferencesQueries.TableColumns
+import explore.model.AppContext
+import explore.model.ColumnId
+import explore.model.TableColumnPref
+import explore.model.enums.SortDirection
+import explore.model.enums.TableId
+import explore.model.reusability.given
+import explore.syntax.ui.*
+import japgolly.scalajs.react.Callback
+import japgolly.scalajs.react.CallbackTo
+import japgolly.scalajs.react.hooks.CustomHook
+import lucuma.core.model.User
+import lucuma.react.table.ColumnDef
+import reactST.tanstackTableCore.mod.Table
+import reactST.{tanstackTableCore => raw}
+
+import scalajs.js.JSConverters.*
+import scalajs.js
+
+trait TableHooks:
+  def toSortingRules(tuples: List[(String, Boolean)]): js.Array[raw.mod.ColumnSort] =
+    tuples.map { case (id, b) => raw.mod.ColumnSort(b, id) }.toJSArray
+
+  def fromTableState(state: raw.mod.TableState): List[(String, Boolean)] =
+    state.sorting.toList.map(sr => (sr.id.toString, sr.desc))
+
+  def useTablePreferencesLoad[T] =
+    CustomHook[
+      (Option[User.Id], AppContext[IO], TableId, List[ColumnDef[T, ?]], List[TableColumnPref])
```

Comment:
Can we use a `case class` for the options? That will name the parameters, and the context can be given.

Reply:
Sure. I avoided it to skip creating an object on call but it is probably clearer to have it